kotlin-native/litmus/src/commonMain/kotlin/ChiSquareTest.kt

```diff
@@ -0,0 +1,111 @@
+// returns true if samples have the same distribution with 0.05 significance level
+// written almost completely by ChatGPT :)
+fun chiSquaredTest(sample1: List<Int>, sample2: List<Int>): Boolean {
+    require(sample1.size == sample2.size) { "incomparable samples" }
+    val n = sample1.size // number of categories
+    val dof = n - 1 // degrees of freedom
+    val obs = Array(2) { IntArray(n) } // observed frequencies
+
+    // fill in observed frequencies
+    for (i in 0 until n) {
+        obs[0][i] = sample1[i]
+        obs[1][i] = sample2[i]
+    }
+
+    // calculate expected frequencies
+    val total1 = sample1.sum()
+    val total2 = sample2.sum()
+    val exp = Array(2) { DoubleArray(n) }
+    for (i in 0 until n) {
+        exp[0][i] = total1.toDouble() * (sample1[i] + sample2[i]) / (total1 + total2)
+        exp[1][i] = total2.toDouble() * (sample1[i] + sample2[i]) / (total1 + total2)
+    }
+
+    // calculate chi-squared test statistic
+    var chiSq = 0.0
+    for (i in 0 until n) {
+        for (j in 0 until 2) {
+            val o = obs[j][i].toDouble()
+            val e = exp[j][i]
+            chiSq += (o - e) * (o - e) / e
+        }
+    }
+
+    val targetChiSq = chiSquaredDist[dof]
+    return chiSq < targetChiSq
+}
+
+fun chiSquaredTest(samples: List<List<Int>>): Boolean {
```

Comment:
It would make sense to make classes or typealiases for `List<List<Int>>`, something like `Sample` or `Probe`, etc (please check up literature, other libraries or ChatGPT for a suitable name).